### ISSUE TITLE
[HL] Some Weapons do not Play Idle Animations

### ISSUE THREAD

**Issue by Nevermore2790** [ValveSoftware/halflife#1585]
Weapons like: Crowbar, Gluon Gun and Tau Cannon have idle animations (can be seen in the model viewer, Half-Life: Source and some mods) that do not play in the final release. This thing haven't been fixed since the game's release up to now.

**Comment by HARDSTYLEBG** [comment]
Yep one of the gauss animation for example is rotating the gauss spinning part with fingers on the left hand

**Comment by Nevermore2790** [comment]
@HARDSTYLEBG - Don't forget about the Gluon Gun's flare effect too, and looking at the model viewer, the Crowbar has 3 idle animations (only 1 is used). It also included an unused attack hit animation (this must be the attack animation of the Crowbar in Black Mesa is based on). Same with the Tau Cannon, it has 2 firing animations (the other firing animation is the only firing animation in Half-Life: Source is used ---- looking at it, I am curious to which attack is that unused one. Either primary or secondary?)

**Comment by HARDSTYLEBG** [comment]
@Nevermore2790 Yes i know about all of them (: I just gave only this as example.This video shows some of the flaws of the new Half-Life https://www.youtube.com/watch?v=uKkJPFmDW9Q ;I think those animations and other stuff are cutted out to reduce size because back then the internet was slow anyway now most people have decent internet so i don't think there will be any problem to put them back.

**Comment by JoelTroch** [comment]
I made the following tests on a clean HL1's source code, this means that it does not concern the retail Half-Life but the mods, I was using the Blue-Shift/Opposing Force HD pack :
* Crowbar's idle animations seems to be not played.
* All Python's idle animations are played, only the "reload" sound is missing.
* The MP5's fire rate seems to be the same as the old HL's version.
* No transparency on Gauss gun and Gluon gun beams.
* No "end portal" effect on Gluon gun.
* Hivehand's deploy animation doesn't have time to finish.
* I was able to shoot the dead Barney on all his body part, for some of them (especially the legs), I had to be at an "exact" position otherwise the bullets goes to the world.
* The "fast crowbar corpse gibber" is also present.

I can make some fixes for the things reported above, but remember, it will be for HL1 mods, it's up to Valve to integrate them in the standard Half-Life.

**Comment by Nevermore2790** [comment]
@Joel - In the SteamPipe Half-Life 1, the MP5's fire rate is now the same as the old version. 357 already has a reload sound (except for HD, OpFor and BS) since the first Steam Version. Also, where's the Tau Cannon and Gluon Gun?

**Comment by JoelTroch** [comment]
@Nevermore2790 : I was forced to clean my fork because it's a huge mess (I'm not a Git expert).

For the 357 sound, I think adding a sound event to the MDL is fine, no need to have a server side sound.
I didn't found yet for the Tau and Gauss, but I'll keep looking when I have time.

**Comment by Nevermore2790** [comment]
Where's that .cpp file in the game files?? I can't seem to find it?

**Comment by JoelTroch** [comment]
I sent a more clean PR for the Crowbar's idle animation thanks to some Git/GitHub training.

@Nevermore2790 : CPP files are for the source code, you won't find them in game's files.

**Comment by Nevermore2790** [comment]
Can't find that Source Code easily, so I might just need some assistance or a download or someone try to fix it. :)

**Comment by JoelTroch** [comment]
### If you DON'T have Git installed (all OS)

Go the main [repository page](https://github.com/ValveSoftware/halflife), on the right, you should see a **"Download ZIP"** button, click it, let GitHub generate the ZIP file for you, then your favorite Internet browser will handle the rest.

### If you are running on Windows and have [Git for Windows](https://msysgit.github.io/) installed

Go to the folder where you want the source code to be downloaded, then hold **SHIFT** and right click in an empty space in Windows's explorer. You should see **"Git Bash"** in the menu, click on it to open it, then clone this repository using this command:

    git clone https://github.com/ValveSoftware/halflife.git

If you don't like commands, you can use a GUI (Graphical User Interface) client, I'm using [GitExtensions](https://code.google.com/p/gitextensions/), but GitHub has it's own [client](https://windows.github.com/) and there is also [SourceTree](https://www.sourcetreeapp.com/).

### If you are running on Linux/Mac OS and have Git installed

You just need to open the terminal of your Linux distro/Mac OS then type the same command as Windows which is:

    git clone https://github.com/ValveSoftware/hallife.git

**Comment by JoelTroch** [comment]
I think I know why Valve disabled the Gauss gun's spin animation (called ``fidget`` in the MDL file), have a look below (remember it's a theory):

Game/Mod | Low Definition (LD) Model | High Definition (HD) Model
-------- | -------------------------- | ---------------------------
Half-Life | Normal | No hands, just a spin and a little move
Blue Shift | Same MDL as Half-Life LD | Same MDL as Half-Life HD
Opposing Force | No MDL, so it uses the one from Half-Life LD | Same MDL as Half-Life HD

I also found that idle animations will be played only if you overcharge the Gauss gun.

**Comment by Nevermore2790** [comment]
I do not have Git installed, and I have downloaded it (file is halflife-master). I can now see the Source Code of the weapons... okay, I really can't understand what to do unless if there are pictures.... :)
Gearbox, really did not edit the Tau Cannon (because in the final release of Half-Life 1, it's frozen in idle). 
Also, it can also play its idle animation upon a loading screen, but sadly, it only uses 1 idle animation.

**Comment by agrastiOs** [comment]
@kisak-valve, this issue details the same problems #2495, but the latter details the issue much better and includes fixes, so I suggest this gets closed.
